pkg/service/catalog/api/types.go

```diff
@@ -0,0 +1,21 @@
+package api
+
+import (
+	kapi "k8s.io/kubernetes/pkg/api"
+)
+
+type CatalogService struct {
+	// Target is a reference to a service
```

Comment:
You'd `oc get catalogservices/foo` to get all the services in catalog foo. I don't anticipate needing to retrieve one by name, but maybe? We will want to filter based on labels, so @ingvagabund go ahead and add ObjectMeta.

I guess let's add TypeMeta too. It can't hurt anything.